packages/nx-spawn/src/index.ts

```diff
@@ -1,107 +1,39 @@
 #!/usr/bin/env node
-import { exit, cwd, argv } from 'process';
-import { join } from 'path';
-import {
-  createProjectGraphAsync,
-  workspaceRoot,
-  type ProjectGraph,
-} from '@nrwl/devkit';
-/* eslint-disable import/extensions */
-import { Workspaces } from 'nx/src/config/workspaces.js';
-/* eslint-enable import/extensions */
-import concurrently from 'concurrently';
+import { workspaceRoot, Workspaces } from '@nrwl/devkit';
+import { argv, cwd, exit } from 'process';
 import yargs from 'yargs';
 import { hideBin } from 'yargs/helpers';
-
-interface CommandInfo {
-  packageName: string;
-  cwd: string;
-}
-
-function computeDepCommands(
-  projectGraph: ProjectGraph,
-  parentPackage: string,
-  command: string
-): CommandInfo[] {
-  const deps = projectGraph.dependencies[parentPackage];
-  if (!deps) return [];
-  return (
-    deps
-      // A dependency may or may not be one of our packages - if it is, projectGraph.nodes will include it
-      .filter((dep) => Object.keys(projectGraph.nodes).includes(dep.target))
-      .map((dep) => {
-        const data = projectGraph.nodes[dep.target]?.data as unknown;
-        if (typeof data !== 'object' || data === null)
-          throw new Error('Unable to determine dependency directory');
-        const { root } = data as Record<string, unknown>;
-        if (!root || typeof root !== 'string')
-          throw new Error(`Root directory for ${dep.target} not found`);
-        const resolvedRoot = join(workspaceRoot, root);
-
-        return [
-          ...computeDepCommands(projectGraph, dep.target, command),
-          {
-            packageName: dep.target,
-            cwd: resolvedRoot,
-          },
-        ];
-      })
-      .flat()
-  );
-}
+import TaskOrchestrator from './task-orchestrator';
 
 async function run() {
```

Comment:
I'm not sure I understand the purpose of nx-spawn. What is it used for?

Reply:
The reason why this package exists is because nx is currently unable to handle spinning up watch mode dependencies. For example, consider you have an app that relies on a library. If we compile the app in watch mode, nx will trigger the dependency to compile in watch mode, but because watch mode never exits, nx just hangs and doesn't compile our app (because it's waiting for the dependency build to complete, which will never happen).

This package will be able to go away entirely once https://github.com/nrwl/nx/pull/10706 gets merged